pkg/controller/client/controller_client.go

```diff
@@ -306,13 +307,14 @@ func (c *ControllerClient) ReplicaUpdate(address string, mode types.Mode) (*type
 	return GetControllerReplicaInfo(cr), nil
 }
 
-func (c *ControllerClient) ReplicaPrepareRebuild(address string) ([]types.SyncFileInfo, error) {
+func (c *ControllerClient) ReplicaPrepareRebuild(address, instanceName string) ([]types.SyncFileInfo, error) {
```

Comment:
Can we can also add instanceName to [ReplicaCreate](https://github.com/longhorn/longhorn-engine/blob/0504cf983e5224b116368117b74bc358257da678/pkg/controller/client/controller_client.go#L263) for verification?

Reply:
I think not without modifying the controller's in-memory structures (as discussed above) and doing some in-depth modification of gRPC messages. 

`ReplicaPrepareRebuild` directly creates a replica client (which can make use of the instanceName) and its gRPC call takes a ReplicaAddress (which was easily modified to include an instanceName field).

`ReplicaCreate` creates a `Backend` structure and then uses it to facilitate communication with the replica. While adding an additional instanceName field to the `Remote` type of `Backend` is fairly easy, the `ReplicaCreate`gRPC call itself takes an `Address` (string) instead of a `ReplicaAddress` (modified with an instanceName field), so we would likely want to modify this (and similar requests for consistency).

In short, I think this work crosses the line we discussed previously. If we feel it is necessary, I propose we do it in a followup series of PRs.

Note that ReplicaCreate is only called in two places. In both places, we make a direct call to a replica with the instanceName BEFORE calling ReplicaCreate, so I don't think there's actually a practical benefit to modifying it.

- https://github.com/longhorn/longhorn-engine/blob/bee235337246b50b5c3dd86e46cfebdef3fe3434/pkg/sync/sync.go#L416-L428
- https://github.com/longhorn/longhorn-engine/blob/bee235337246b50b5c3dd86e46cfebdef3fe3434/pkg/sync/sync.go#L362-L372

Reply:
I see. It sounds good to me now. Let's keep the current implementation